src/Storages/StorageFuzzJSON.cpp

```diff
@@ -0,0 +1,511 @@
+        node->fixed = std::move(field);
+    }
+}
+
+std::shared_ptr<Node> parseJSON(const String & json)
+{
+    std::string_view view{json.begin(), json.end()};
+    ParserImpl::Element document;
+    ParserImpl p;
+
+    if (!p.parse(json, document))
+        throw Exception(ErrorCodes::INCORRECT_DATA, "Failed to parse JSON");
+
+    auto root = std::make_shared<Node>();
+    traverse(document, root);
+    return root;
+}
+
+char generateRandomCharacter(pcg64 & rnd, const char * charset, size_t charset_size)
+{
+    auto idx = uniform(0, charset_size - 1)(rnd);
+    return charset[idx];
+}
+
+char generateRandomKeyCharacter(pcg64 & rnd)
+{
+    static constexpr const char charset[] = "0123456789"
+                                            "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                            "abcdefghijklmnopqrstuvwxyz";
+    return generateRandomCharacter(rnd, charset, sizeof(charset) - 1);
+}
+
+char generateRandomStringValueCharacter(pcg64 & rnd)
+{
+    static constexpr const char charset[] = "0123456789"
+                                            "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                            "abcdefghijklmnopqrstuvwxyz"
+                                            "!@#$%^&*-+_";
+    return generateRandomCharacter(rnd, charset, sizeof(charset) - 1);
+}
+
+String generateRandomStringValue(UInt64 min_length, UInt64 max_length, pcg64 & rnd)
+{
+    size_t size = min_length + rnd() % (max_length - min_length + 1);
+    String res;
+    res.reserve(size);
+    for (size_t i = 0; i < size; ++i)
+        res.push_back(generateRandomStringValueCharacter(rnd));
+
+    return res;
+}
+
+String generateRandomKey(UInt64 min_length, UInt64 max_length, pcg64 & rnd)
+{
+    size_t size = min_length + rnd() % (max_length - min_length + 1);
+    String res;
+    res.reserve(size);
+    for (size_t i = 0; i < size; ++i)
+        res.push_back(generateRandomKeyCharacter(rnd));
+
+    return res;
+}
+
+Field generateRandomFixedValue(const StorageFuzzJSON::Config & config, pcg64 & rnd)
+{
+    // TODO (@jkartseva): support more field types.
+    static std::array<Field::Types::Which, 3> possible_types{
+        Field::Types::Which::UInt64, Field::Types::Which::String, Field::Types::Which::Bool};
+
+    Field f;
+    auto idx = rnd() % possible_types.size();
+    switch (possible_types[idx])
+    {
+        case Field::Types::Which::UInt64: {
+            f = rnd();
+            break;
+        }
+        case Field::Types::Which::String:
+            f = generateRandomStringValue(1, config.max_string_value_length, rnd);
+            break;
+        case Field::Types::Which::Bool:
+            f = bool(rnd() % 2);
+            break;
+        default:
+    }
+    return f;
+}
+
+String fuzzJSONKey(const FuzzConfig & config, pcg64 & rnd, const String & source_str)
+{
+    std::list<char> s(source_str.begin(), source_str.end());
```

Comment:
Consider create output string and decide what to put in it in `switch` below (char from source, another char, nothing) instead of using `std::list`.